sonic_y_cable/y_cable.py

```diff
@@ -507,3 +552,537 @@ def check_if_link_is_active_for_torB(physical_port):
+        result = platform_chassis.get_sfp(
+            physical_port).write_eeprom(curr_offset, 1, buffer)
+        if result is False:
+            return result
+        buffer = bytearray([0])
+        curr_offset = Y_CABLE_INITIATE_EYE_MEASUREMENT
+        result = platform_chassis.get_sfp(
+            physical_port).write_eeprom(curr_offset, 1, buffer)
+        if result is False:
+            return result
+
+        while(True):
+            done = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset, 1)
+            y_cable_validate_read_data(done, 1, physical_port, "EYE data ready to read")
+            if done[0] == 1:
+                break
+
+        idx = 0
+        maxLane = 2
+        for lane in range(maxLane):
+            curr_offset = Y_CABLE_LANE_1_EYE_RESULT
+            msb_result = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset+idx, 1)
+            y_cable_validate_read_data(msb_result, 1, physical_port, "EYE data msb result")
+            lsb_result = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset+1+idx, 1)
+            y_cable_validate_read_data(lsb_result, 1, physical_port, "EYE data lsb result")
+            lane_result = (msb_result[0] << 8 | lsb_result[0])
+            eye_result.append(lane_result)
+            idx += 2
+
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to configure the PRBS type")
+        return -1
+
+    return eye_result
+
+
+@hook_y_cable_simulator
+def get_pn_number_and_vendor_name(physical_port):
+    """
+    This API specifically returns the pn number and vendor name for a specfic port.
+
+    Args:
+        physical_port:
+             an Integer, the actual physical port connected to a Y cable
+    Returns:
+        a tuple, with pn_number and vendor name
+    """
+
+    curr_offset = Y_CABLE_PN_NUMBER
+
+    if platform_chassis is not None:
+        pn_result = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset, 15)
+        y_cable_validate_read_data(pn_result, 1, physical_port, "PN number")
+        curr_offset = Y_CABLE_VENDOR_NAME
+        vendor_name = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset, 15)
+        y_cable_validate_read_data(vendor_name, 15, physical_port, "vendor name")
+
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to get pin results")
+        return -1
+
+    return pn_result, vendor_name
+
+
+@hook_y_cable_simulator
+def get_switch_count(physical_port, count_type):
+    """
+    This API specifically returns the switch count to change the Active TOR which has
+    been done manually by the user.
+
+    Args:
+        physical_port:
+             an Integer, the actual physical port connected to a Y cable
+        count_type:
+             a string, for getting the count type
+                      "manual" -> manual switch count
+                      "auto" -> automatic switch count
```

Comment:
Suggest creating constants for these options, also (e.g., `SWITCH_TYPE_MANUAL`, `SWITCH_TYPE_AUTO`)

Reply:
fixed